### PR TITLE
Pathway analysis: QuSAGE for single dataset

### PR DESCRIPTION
**Pull request by jaclyn-taroni**
#33 - single dataset example

Here, I'm using [`GSE75574`](https://www.refine.bio/experiments/GSE75574/gene-expression-in-mouse-tissues-in-response-to-short-term-calorie-restriction) as an example. This is a SuperSeries that examines the transcriptome in response to short-term calorie restriction in multiple tissues and multiple mouse strains. 

In this notebook, I use the gene sets introduced in #44 to perform pathway analysis with QuSAGE. I only use white adipose tissues and put all strains together (not used as a covariate). Other steps I've included along the way: conversion from Ensembl gene ID to gene symbols, making a heatmap with only genes in a particular pathway of interest.

### CODE REVIEW

File: pathway-analysis/qusage_single_dataset.Rmd [hunk 1]
```diff
@@ -0,0 +1,420 @@
...
+
+Retain only the most pertinent columns.
+
+```{r}
+metadata_df <- metadata_df %>%
+  dplyr::select(refinebio_accession_code, refinebio_title, 
+                refinebio_specimen_part, refinebio_sex, strain)
+```
+
+This particular accession (`GSE75574`) is a SuperSeries comprised of experiments
```
Comment (cansav09) [inline pathway-analysis/qusage_single_dataset.Rmd]: Could put link to the refine.bio page so the info is readily accessible.

File: pathway-analysis/qusage_single_dataset.Rmd [hunk 2]
```diff
@@ -0,0 +1,420 @@
...
+in multiple tissues from multiple mouse strains. 
+We're interested in white adipose tissue for all strains.
```
Comment (cansav09) [inline pathway-analysis/qusage_single_dataset.Rmd]: What's the reasoning for why we are interested in this?

File: pathway-analysis/qusage_single_dataset.Rmd [hunk 3]
```diff
@@ -0,0 +1,420 @@
...
+
+We can get a look at the general trend of the results with the `plotCIs` 
+function. 
+This plots the means and 95% confidence intervals of each pathway we tested, 
+sorted such that the gene sets with the highest mean will be on the left of the 
+plot.
+These are gene sets that are elevated in calorie restricted mice. 
+Error bars are colored by the directionality and corrected p-value (FDR by
+default).
+Unfortunately the p-value color scheme is red-green.
```
Comment (cansav09) [inline pathway-analysis/qusage_single_dataset.Rmd]:
```suggestion
Unfortunately the p-value color scheme is red-green, which is generally not colorblindness friendly. 
```
Some people might think you just hate the colors red and green.

File: pathway-analysis/qusage_single_dataset.Rmd [hunk 4]
```diff
@@ -0,0 +1,420 @@
...
+* It returns more information than just a p-value. 
+That's useful for analyses you might want to perform downstream.
+* Built-in visualization functionality.
+
+We recommend taking a look at the original publication (Yaari et al.) and 
+the R package documentation to learn more.
+
+## Gene sets
+
+Here we'll work with a mouse dataset.
```
Comment (cansav09) [inline pathway-analysis/qusage_single_dataset.Rmd]: This sentence feels a bit abrupt and maybe out of place. Maybe put it in the next paragraph?

File: pathway-analysis/qusage_single_dataset.Rmd [hunk 5]
```diff
@@ -0,0 +1,420 @@
...
+Save the `QSArray` output.
+We can use this object for visualization or other downstream analyses.
+
+```{r}
+readr::write_rds(qusage_results,
+                 file.path("results", 
+                           "GSE75574_adipose_calorierest-ctrl_QSarray.RDS"))
+```
+
+### Overall
```
Comment (cansav09) [inline pathway-analysis/qusage_single_dataset.Rmd]:
```suggestion
### Overall Results
```
Overall of what?

**Comment by jaclyn-taroni** [comment]
Thanks @cansav09 - made some wording changes! I'm going to merge this now even though it's long (#53 is also long). We can have @dvenprasad go over the module after everything gets added and give us notes.

**Pull request merged**